src/main/java/org/sagebionetworks/bridge/exporter/record/RecordIdSourceFactory.java

```diff
@@ -83,64 +74,34 @@ final void setS3Helper(S3Helper s3Helper) {
      * @throws IOException
      *         if we fail reading the underlying source
      */
-    public Iterable<String> getRecordSourceForRequest(BridgeExporterRequest request) throws IOException {
+    public Iterable<String> getRecordSourceForRequest(BridgeExporterRequest request, DateTime endDateTime, Map<String, DateTime> studyIdsToQuery) throws IOException {
         if (StringUtils.isNotBlank(request.getRecordIdS3Override())) {
             return getS3RecordIdSource(request);
-        } else if (request.getStudyWhitelist() != null) {
-            return getDynamoRecordIdSourceWithStudyWhitelist(request);
         } else {
-            return getDynamoRecordIdSource(request);
+            return getDynamoRecordIdSourceGeneral(endDateTime, studyIdsToQuery);
         }
     }
 
     /**
-     * If we have a study whitelist, we should always use the study-uploadedOn index, as it's more performant. If we
-     * have a date, we'll need to compute the start and endDateTimes based off of that. Otherwise, we use the start-
-     * and endDateTimes verbatim.
+     * Helper method to get ddb records
+     * @return
      */
-    private Iterable<String> getDynamoRecordIdSourceWithStudyWhitelist(BridgeExporterRequest request) {
-        // Compute start- and endDateTime.
-        DateTime startDateTime;
-        DateTime endDateTime;
-        if (request.getDate() != null) {
-            // startDateTime is obviously date at midnight local time. endDateTime is the start of the next day.
-            LocalDate date = request.getDate();
-            startDateTime = date.toDateTimeAtStartOfDay(timeZone);
-            endDateTime = date.plusDays(1).toDateTimeAtStartOfDay(timeZone);
-        } else {
-            // Logically, if there is no date, there must be a start- and endDateTime. (We check for recordIdS3Override
-            // in the calling method.
-            startDateTime = request.getStartDateTime();
-            endDateTime = request.getEndDateTime();
-        }
-
-        // startDateTime is inclusive but endDateTime is exclusive. This is so that if a record happens to fall right
-        // on the overlap, it's only exported once. DDB doesn't allow us to do AND conditions in a range key query,
-        // and the BETWEEN is always inclusive. However, the granularity is down to the millisecond, so we can just
-        // use endDateTime minus 1 millisecond.
-        long startEpochTime = startDateTime.getMillis();
-        long endEpochTime = endDateTime.getMillis() - 1;
-
-        // RangeKeyCondition can be shared across multiple queries.
-        RangeKeyCondition rangeKeyCondition = new RangeKeyCondition("uploadedOn").between(startEpochTime,
-                endEpochTime);
+    private Iterable<String> getDynamoRecordIdSourceGeneral(DateTime endDateTime, Map<String, DateTime> studyIdsToQuery) {
+        // proceed
+        Iterable<Item> recordItemIter;
```

Comment:
Minor: You don't need to declare this variable at the top when it's not used until the end.